beacon_node/network/src/sync/manager.rs

```diff
@@ -567,48 +688,119 @@ impl<T: BeaconChainTypes> SyncManager<T> {
+                        peer_id,
+                        &mut self.network,
+                    );
+                    if self.should_delay_lookup(blob_slot) {
+                        if let Err(e) = self
+                            .delayed_lookups
+                            .try_send(SyncMessage::BlobParentUnknown(peer_id, blob))
+                        {
+                            warn!(self.log, "Delayed lookups dropped for blob"; "block_root" => ?block_root, "error" => ?e);
+                        }
+                    } else {
+                        let mut blobs = FixedBlobSidecarList::default();
+                        *blobs.index_mut(blob_index as usize) = Some(blob);
+                        self.block_lookups
+                            .search_current_unknown_parent_block_and_blobs(
+                                block_root,
+                                None,
+                                Some(blobs),
+                                &[PeerShouldHave::Neither(peer_id)],
+                                &mut self.network,
+                            );
+                    }
                 }
             }
-            SyncMessage::UnknownBlockHash(peer_id, block_hash) => {
+            SyncMessage::MergedParentUnknown(block_root, peers, block, blobs) => {
+                self.block_lookups
+                    .search_current_unknown_parent_block_and_blobs(
+                        block_root,
+                        block,
+                        blobs,
+                        peers.as_slice(),
+                        &mut self.network,
+                    );
+            }
+            SyncMessage::UnknownBlockHashFromAttestation(peer_id, block_hash) => {
                 // If we are not synced, ignore this block.
-                if self.network_globals.sync_state.read().is_synced()
-                    && self.network_globals.peers.read().is_connected(&peer_id)
-                    && self.network.is_execution_engine_online()
-                {
-                    self.block_lookups
-                        .search_block(block_hash, peer_id, &mut self.network);
+                if self.synced_and_connected(&peer_id) {
+                    self.block_lookups.search_block(
+                        block_hash,
+                        PeerShouldHave::BlockAndBlobs(peer_id),
+                        &mut self.network,
+                    );
                 }
             }
-            SyncMessage::UnknownBlobHash { .. } => {
-                unimplemented!()
+            SyncMessage::MissingGossipBlockComponents(slot, peer_id, block_hash) => {
+                // If we are not synced, ignore this block.
+                if self.synced_and_connected(&peer_id) {
+                    if self.should_delay_lookup(slot) {
+                        if let Err(e) = self.delayed_lookups.try_send(
+                            SyncMessage::UnknownBlockHashFromAttestation(peer_id, block_hash),
```

Comment:
Might need a different message here? Should be nicer on moving `delayed_lookups` to its own module with it's own message types.

Reply:
Yea I actually think this was a bug and should have passed along `MissingGossipBlockComponents`, but you're right a new message will be better, adding that.